### DISCUSSION TITLE
image-reflector-controller returns "no tags in database"

### DISCUSSION THREAD

**Discussion by ViacheslavKudinov** [fluxcd/flux2#4887] [Q&A]
Hello,
`image-reflector-controller` started to return the error:
```
{"level":"error","ts":"2024-07-14T09:14:24.554Z","msg":"no tags in database","controller":"imagepolicy","controllerGroup":"image.toolkit.fluxcd.io","controllerKind":"ImagePolicy","ImagePolicy":{"name":"CENSORED","namespace":"flux-CENSORED"},"namespace":"flux-CENSORED","name":"CENSORED","reconcileID":"86083478-b403-49ab-8e94-3d3da00ce17c","error":"DependencyNotReady"}
```
No changes as we know on imagepolicy definition, no restarts of the controller.

`Flux version 2.2.2`

More info about controller:
```
ghcr.io/fluxcd/image-reflector-controller:v0.31.1

    Args:
      --events-addr=http://notification-controller.flux-system.svc.cluster.local./
      --watch-all-namespaces=true
      --log-level=info
      --log-encoding=json
      --enable-leader-election
```

We couldn't find yet anything about it on documentation. 

Any tips or advice is appreciated.

**Comment by ViacheslavKudinov** [comment]
I've found the root cause.

**Reply by RonnanSouza** [reply]
What was the root cause, @ViacheslavKudinov ??

**Reply by ViacheslavKudinov** [reply]
It was expected behaviour as images were deleted from Nexus image registry. They were for keeping snapshot images and after some time were cleaned up by scheduler.

**Comment by nicks** [comment]
We had a bug where the `image-reflector-controller` was getting rescheduled a lot. 

And one of the symptoms was that we would see this event a ton.

I guess it's expected to see this error message during image-reflector-controller startup when it's populating its DB?

**Reply by ViacheslavKudinov** [reply]
> We had a bug where the `image-reflector-controller` was getting rescheduled a lot.
> 
> And one of the symptoms was that we would see this event a ton.
> 
> I guess it's expected to see this error message during image-reflector-controller startup when it's populating its DB?

We see this errors every time when controller starts, but it is automatically resolved after minute or so.
I'm mot sure it has to generate this error on each start up, but it is how it is for now.
